src/routing/enduser_registry.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright 2021 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use crate::error::{Error, Result};
+use sn_data_types::{PublicKey as EndUserPK, Signature as EndUserSig};
+use sn_messaging::EndUser;
+use std::{
+    collections::{btree_map::Entry, BTreeMap},
+    net::SocketAddr,
+};
+use xor_name::XorName;
+
+pub type SocketId = XorName;
+pub(crate) struct EndUserRegistry {
+    clients: BTreeMap<SocketAddr, EndUser>,
+    socket_id_mapping: BTreeMap<SocketId, SocketAddr>,
+}
+
+impl EndUserRegistry {
```

Comment:
Do we eveer remove stuff from this registry?

Reply:
It would be removed on ClientLost. But I suggest that flow be added in a new PR, how's that sound?